### PR TITLE
Feature/remove shared networks

### PR DESCRIPTION
**Pull request by DriesSmit**
## What?
Remove the shared policy and critic network option from IPPO.
## Why?
To simplify the code, we will only be evaluating IPPO using separate networks for the JAX release. If necessary, we can add the shared networks code back later.
## How?
Remove all code associated with shared networks IPPO. Renamed the separate networks components by removing the shared networks tag.

### CODE REVIEW

**Review by AsadJeewa — APPROVED** [review]
Thanks @DriesSmit 
Everything runs on my side, incl the tests. Thanks for also changing it to OpimiSer

File: examples/jax/quickstart.ipynb [hunk 1]
```diff
@@ -118,7 +118,7 @@
         "id": "l8XqA9M2iyK_"
       },
       "source": [
-        "For this example will train IPPO on the simple-spread environment. This environment has 3 agents that all need to move to specific locations without bumping into each other. We start by defining the network architecture and network optimizer for our agents. The default network file for ippo can be found [here](https://github.com/instadeepai/Mava/blob/develop/mava/systems/jax/ippo/networks.py)."
+        "For this example will train IPPO on the simple-spread environment. This environment has 3 agents that all need to move to specific locations without bumping into each other. We start by defining the network architecture and network optimisers for our agents. The default network file for ippo can be found [here](https://github.com/instadeepai/Mava/blob/develop/mava/systems/jax/ippo/networks.py)."
```
Comment (RuanJohn) [inline examples/jax/quickstart.ipynb:121]: For this example we will

File: mava/systems/jax/ippo/networks.py [hunk 2]
```diff
@@ -312,6 +160,104 @@ def make_ppo_networks(
...
+            should be used.
+            Defaults to True.
+
+    Returns:
+        PPONetworks class
+    """
+
+    num_actions = environment_spec.actions.num_values
+
+    # TODO (dries): Investigate if one forward_fn function is slower
```
Comment (RuanJohn) [inline mava/systems/jax/ippo/networks.py:212]: We can probably remove this TODO now?

**Review by RuanJohn — COMMENTED** [review]
Thank you for all the effort here @DriesSmit! Just some minor comments from my side.

File: mava/systems/jax/ippo/networks.py (hunk 2 above)
Comment (DriesSmit) [reply mava/systems/jax/ippo/networks.py]: Removed :+1:

File: examples/jax/quickstart.ipynb (hunk 1 above)
Comment (DriesSmit) [reply examples/jax/quickstart.ipynb]: Fixed :)

Comment (AsadJeewa) [reply examples/jax/quickstart.ipynb]: "we" is still missing

File: mava/systems/jax/ippo/networks.py (hunk 2 above)
Comment (RuanJohn) [reply mava/systems/jax/ippo/networks.py]: Thank you :)

**Review by AsadJeewa — APPROVED** [review]

**Review by RuanJohn — APPROVED** [review]
Thank you @DriesSmit it makes things a lot cleaner!

**Pull request merged**